recipes/solexaqa/meta.yaml

```diff
@@ -0,0 +1,48 @@
+{% set name = "solexaqa" %}
+{% set version = "3.1.7.1" %}
+{% set sha256 = "b3bdabf955387a4f6176dbdfda1cf03bb869c077b4eec152304f2e76be5f0cf6" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+build:
+  number: 0
+  skip: True # [osx or win]
+
+source:
+  url: https://downloads.sourceforge.net/project/solexaqa/src/SolexaQA++_v{{ version }}.zip
+  sha256: {{ sha256 }}
+
+requirements:
+  build:
+    - gxx_linux-64 >=7.3.0
+    - make
+  host:
+    - boost
+    - icu
+    - libstdcxx-ng
+    - zlib
+  run:
+    - boost
+    - icu
+    - libstdcxx-ng
+    - zlib
+    # R is not a strict dependency, but can provide enhanced functionality,
+    # so we leave it as an option for the user, not a hard requirement.
+    # - r
+
+test:
+  files:
+    - tiny.fastq
+  commands:
+    - SolexaQA++ dynamictrim tiny.fastq >/dev/null
```

Comment:
/dev/null can be removed.
Is the tiny.fastq file part of the pkg or only part of the tarball from which you install? If its part of the package you could `which SolexaQA++`\share\tiny.fastq the file maybe.

Reply:
Not part of the package.  It's a trivial test file for exercising the application.  I'm currently struggling to get it into the test environment, because of the lack of support from mulled-build for test::files.  My current thinking is 4 echo commands, since it's a 4 line file.  Ugh!

Reply:
Never tried it, but in theory
```yaml
test:
  commands:
    - |
      SolexaQA++ dynamictrim <(echo -n '@HWI-ST871:91:C0JGJACXX:8:1302:10862:157183 1:N:0:TAGCTT
      TGTCTCTGCTTTTGAATATGCTATCTAGGTTGGTCATAACTTTCTTTCCAAGGAGTAATCATCTTTTAATTTCATGGCTACAATCACCATTTGCAGTTATT
      +
      CCCCFFFFFHHHHHJIJIJJJIIJJIGJJIJIJHIGHIJJJJJJJJIJJJJIIIGFEGIGGIJIIIJI>FHIJIHHJIHIHG>CEFFFFFED@C:>AC@>;
      ')
```
could work.
Or a little simpler
```yaml
test:
  commands:
    - |
      echo -n '@HWI-ST871:91:C0JGJACXX:8:1302:10862:157183 1:N:0:TAGCTT
      TGTCTCTGCTTTTGAATATGCTATCTAGGTTGGTCATAACTTTCTTTCCAAGGAGTAATCATCTTTTAATTTCATGGCTACAATCACCATTTGCAGTTATT
      +
      CCCCFFFFFHHHHHJIJIJJJIIJJIGJJIJIJHIGHIJJJJJJJJIJJJJIIIGFEGIGGIJIIIJI>FHIJIHHJIHIHG>CEFFFFFED@C:>AC@>;
      ' | SolexaQA++ dynamictrim
```
if `SolexaQA++ dynamictrim` is also able to read from stdin.
If that doesn't work, a less elegant way would be to put this all into one line and use `printf` instead of `echo` and replace all linebreaks by `\n`.

Reply:
Solved with an inline printf.  (But support for test::files would be a really useful addition!)